src/middlewared/middlewared/plugins/iscsi.py

```diff
@@ -1588,15 +1588,20 @@ async def is_child_of_path(self, resource, path):
         return is_child(resource[self.path_field], os.path.join('zvol', os.path.relpath(path, '/mnt')))
 
     async def delete(self, attachments):
-        lun_ids = []
+        orphan_targets_ids = set()
         for attachment in attachments:
             for te in await self.middleware.call('iscsi.targetextent.query', [['extent', '=', attachment['id']]]):
+                orphan_targets_ids.add(te['target'])
```

Comment:
How can you know there are no more extents associated with this target?

Reply:
I discard the ones still associated with some extent later https://github.com/freenas/freenas/pull/5741/files#diff-a6a523dfb3e8a62b0d0efc1730d014c6R1601

Reply:
Ah thanks